content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,1002 @@
+Because this function accesses the internet it needs to be run with the `Runner`
+network policy.
+
+```python
+import sys
+
+import requests
+import yaml
+
+ANNOTATION_KEY_AUTHOR = "quotable.io/author"
+ANNOTATION_KEY_QUOTE = "quotable.io/quote"
+
+
+def get_quote() -> tuple[str, str]:
+    """Get a quote from quotable.io"""
+    rsp = requests.get("https://api.quotable.io/random")
+    rsp.raise_for_status()
+    j = rsp.json()
+    return (j["author"], j["content"])
+
+
+def read_Functionio() -> dict:
+    """Read the FunctionIO from stdin."""
+    return yaml.load(sys.stdin.read(), yaml.Loader)
+
+
+def write_Functionio(Functionio: dict):
+    """Write the FunctionIO to stdout and exit."""
+    sys.stdout.write(yaml.dump(Functionio))
+    sys.exit(0)
+
+
+def result_warning(Functionio: dict, message: str):
+    """Add a warning result to the supplied FunctionIO."""
+    if "results" not in Functionio:
+        Functionio["results"] = []
+    Functionio["results"].append({"severity": "Warning", "message": message})
+
+
+def main():
+    """Annotate all desired composed resources with a quote from quotable.io"""
+    try:
+        Functionio = read_Functionio()
+    except yaml.parser.ParserError as err:
+        sys.stdout.write("cannot parse FunctionIO: {}\n".format(err))
+        sys.exit(1)
+
+    # Return early if there are no desired resources to annotate.
+    if "desired" not in Functionio or "resources" not in Functionio["desired"]:
+        write_Functionio(Functionio)
+
+    # If we can't get our quote, add a warning and return early.
+    try:
+        quote, author = get_quote()
+    except requests.exceptions.RequestException as err:
+        result_warning(Functionio, "Cannot get quote: {}".format(err))
+        write_Functionio(Functionio)
+
+    # Annotate all desired resources with our quote.
+    for r in Functionio["desired"]["resources"]:
+        if "resource" not in r:
+            # This shouldn't happen - add a warning and continue.
+            result_warning(
+                Functionio,
+                "Desired resource {name} missing resource body".format(
+                    name=r.get("name", "unknown")
+                ),
+            )
+            continue
+
+        if "metadata" not in r["resource"]:
+            r["resource"]["metadata"] = {}
+
+        if "annotations" not in r["resource"]["metadata"]:
+            r["resource"]["metadata"]["annotations"] = {}
+
+        if ANNOTATION_KEY_QUOTE in r["resource"]["metadata"]["annotations"]:
+            continue
+
+        r["resource"]["metadata"]["annotations"][ANNOTATION_KEY_AUTHOR] = author
+        r["resource"]["metadata"]["annotations"][ANNOTATION_KEY_QUOTE] = quote
+
+    write_Functionio(Functionio)
+
+
+if __name__ == "__main__":
+    main()
+```
+
+To build this function you'll need its `requirements.txt` and a `Dockerfile`:
+
+{{< expand "The Function's requirements." >}}
+```
```

Comment:
```suggestion
```python
```
set language for highlighting.

Reply:
I don't think a `requirements.txt` file is Python.

Reply:
It's not, but setting python as the highlighter displays better.
Current:
![Screenshot 2023-01-31 at 5 32 40 PM](https://user-images.githubusercontent.com/10537576/215898900-507cc614-79f6-4697-92f1-29b61c0ce7ad.png)

Setting ` ```python `
![Screenshot 2023-01-31 at 5 32 50 PM](https://user-images.githubusercontent.com/10537576/215898960-3f682a1b-a6b0-41af-beb5-570268cfa0a3.png)

Reply:
Okay, I've changed this.